### PR TITLE
Make level-2

### PR DESCRIPTION
**Pull request by creator-max**
Add Vehicle class. Implement IComporable for Vehicle.

### CODE REVIEW

File: AutoPark/Vehicle.cs [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            Model = model;
+            LicensePlat = licensePlat;
+            Weight = weight;
+            YearOfIssue = yearOfIssue;
+            Mileage = mileage;
+            Color = color;
+            TankVolume = tankVolume;
+        }
+
+        public VehicleType VehicleType { get; }
```
[the remark this answers is not in the document]
Comment (PushMeNow) [reply AutoPark/Vehicle.cs:28]: @kuzmiich I don't see the reason why here need  all accessors. It's correct he is using autoprop.

File: AutoPark/Vehicle.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+        public string Model { get; }
+        public ushort YearOfIssue { get; }
```
[the remark this answers is not in the document]
Comment (PushMeNow) [reply AutoPark/Vehicle.cs:30]: All props have to be initialized by ctor. 'init' accessor useless here

File: AutoPark/Vehicle.cs [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+        public double Weight { get; }
```
[the remark this answers is not in the document]
Comment (PushMeNow) [reply AutoPark/Vehicle.cs:31]: Why they cannot? Weight of vehicle is not editable property.

File: AutoPark/VehicleHelper.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+            var maxMileageVehicle = vehicles[0];
+            foreach (var vehicle in vehicles)
+            {
+                if (vehicle.Mileage > maxMileageVehicle.Mileage)
+                    maxMileageVehicle = vehicle;
+            }
+            return maxMileageVehicle;
+        }
+
+        public static void PrintVehiclesOnConsole(Vehicle[] vehicles)
```
[the remark this answers is not in the document]
Comment (PushMeNow) [reply AutoPark/VehicleHelper.cs:29]: @kuzmiich It is helper for vehicle so generic will destroy whole logic of the helper. I agree with renaming but better PrintVehicles

**Review by PushMeNow — CHANGES_REQUESTED** [review]

File: AutoPark/Vehicle.cs [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+        public double TankVolume { get; }
+        public string LicensePlat { get; set; }
+        public double Mileage { get; set; }
+        public Color Color { get; set; }
+
+        public double GetCalcTaxPerMonth() => Weight * 0.0013 +
```
[the remark this answers is not in the document]
Comment (creator-max) [reply AutoPark/Vehicle.cs]: Thanks, fixed

File: AutoPark/Vehicle.cs [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+            VehicleType.TaxCoefficient * 30 + 5;
+
+        public override string ToString() => $"{Model}, {YearOfIssue}, " +
+            $"{Weight}, {TankVolume}, {LicensePlat}, {Mileage}, " +
+            $"{Color}, {GetCalcTaxPerMonth().ToString("0.00")}";
+
+        public int CompareTo(Vehicle other)
+        {
+            if (other != null)
+                return this.GetCalcTaxPerMonth().CompareTo(other.GetCalcTaxPerMonth());
```
[the remark this answers is not in the document]
Comment (creator-max) [reply AutoPark/Vehicle.cs]: Thanks, fixed

File: AutoPark/VehicleHelper.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (creator-max) [reply AutoPark/VehicleHelper.cs]: Thanks, fixed

**Review by kuzmiich — APPROVED** [review]

**Review by kuzmiich — APPROVED** [review]

**Review by PushMeNow — APPROVED** [review]

**Pull request merged**